adapters/repos/db/helpers/tokenizer.go

```diff
@@ -18,30 +18,41 @@ import (
 	"unicode"
 
 	"github.com/go-ego/gse"
+	koDict "github.com/ikawaha/kagome-dict-ko"
+	kagomeTokenizer "github.com/ikawaha/kagome/v2/tokenizer"
 	"github.com/weaviate/weaviate/entities/models"
 )
 
 var (
 	gseTokenizer     *gse.Segmenter
 	gseTokenizerLock = &sync.Mutex{}
 	UseGse           = false
+	KagomeKrEnabled  = false
 )
 
+// Optional tokenizers can be enabled with an environment variable like:
+// 'ENABLE_TOKENIZER_XXX', e.g. 'ENABLE_TOKENIZER_GSE', 'ENABLE_TOKENIZER_KAGOME_KR'
 var Tokenizations []string = []string{
 	models.PropertyTokenizationWord,
 	models.PropertyTokenizationLowercase,
 	models.PropertyTokenizationWhitespace,
 	models.PropertyTokenizationField,
 	models.PropertyTokenizationTrigram,
-	models.PropertyTokenizationGse,
 }
 
 func init() {
+	if os.Getenv("USE_GSE") == "true" || os.Getenv("ENABLE_TOKENIZER_GSE") == "true" {
+		Tokenizations = append(Tokenizations, models.PropertyTokenizationGse)
+	}
+	if os.Getenv("ENABLE_TOKENIZER_KAGOME_KR") == "true" {
+		Tokenizations = append(Tokenizations, models.PropertyTokenizationKagomeKr)
+	}
 	init_gse()
+	_ = initializeKagomeTokenizerKr()
 }
 
 func init_gse() {
-	if os.Getenv("USE_GSE") == "true" {
+	if os.Getenv("USE_GSE") == "true" || os.Getenv("ENABLE_TOKENIZER_GSE") == "true" {
```

Comment:
to check if value is enabled you can use our helper method [configbase.Enabled](https://github.com/weaviate/weaviate/blob/main/usecases/configbase/config_base.go#L16) then you can use it like [here](https://github.com/weaviate/weaviate/blob/main/usecases/config/environment.go#L41) in this example